templates/base/packages/nextjs/next.config.ts.template.mjs

```diff
@@ -1,24 +1,25 @@
-import { stringify, withDefaults } from '../../../utils.js'
+import { stringify, withDefaults, deepMerge } from '../../../utils.js'
 
-const contents = ({ ignoreTsAndLintBuildErrors, extraConfig }) =>
-`import type { NextConfig } from "next";
-
-const nextConfig: NextConfig = {
-  ${extraConfig[0] ? `...${stringify(extraConfig[0])},` : ''}
+const defaultConfig = {
   reactStrictMode: true,
   devIndicators: false,
   typescript: {
-    ignoreBuildErrors: ${ignoreTsAndLintBuildErrors},
+    ignoreBuildErrors: '$$process.env.NEXT_PUBLIC_IGNORE_BUILD_ERROR === "true"$$',
   },
   eslint: {
-    ignoreDuringBuilds: ${ignoreTsAndLintBuildErrors},
-  },
-  webpack: config => {
-    config.resolve.fallback = { fs: false, net: false, tls: false };
-    config.externals.push("pino-pretty", "lokijs", "encoding");
-    return config;
+    ignoreDuringBuilds: '$$process.env.NEXT_PUBLIC_IGNORE_BUILD_ERROR === "true"$$',
   },
-};
+  webpack: `$$config => { config.resolve.fallback = { fs: false, net: false, tls: false }; config.externals.push("pino-pretty", "lokijs", "encoding"); return config; }$$`,
```

Comment:
yeah this is hacky and also if extension devloper wants to function as value to object key he needs to inline the whole function in one line and add `$${function in one line}$$` 

if we don't do that `stringify` util will change it to : 
https://github.com/scaffold-eth/create-eth/pull/125#discussion_r1804894957

I spend a lot of time changing stringify util to handle this code was getting very very messy and very hacky, so I think this is a fine and simple solution utilising the `$$` convention which we already have 🙌